### DISCUSSION TITLE
How to do `zfs diff`, but in Rust?

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [Inner-Heaven/libzetta-rs#266] [Q&A]
I want to make a backup tool (in Rust) for incrementally backing up ZFS to AWS S3 Glacier Deep storage. For that my plan is to use `zfs diff`. Instead of parsing the output of the `zfs diff` CLI, is there a programmatic way of doing it? Maybe using `libzfs_diff`? If there is a function do do zfs diff could this library include it?

**Comment by andoriyu** [comment]
I don't see support for this in `libzfs_core`. I've started a similar project awhile ago: https://github.com/andoriyu/gazpacho

But I've switched off running ZFS on my desktop, and never have finished it.

IIRC, new version of OpenZFS either already supports JSON output or about to get support for it. This would be the right move.
